packages/jsii-pacmak/lib/targets/golang/types/interface.ts

```diff
@@ -1,55 +1,102 @@
-import { GoType } from './go-type';
-import { InterfaceType, Method, Property, TypeReference } from 'jsii-reflect';
 import { CodeMaker } from 'codemaker';
+import { InterfaceType, Method, Property } from 'jsii-reflect';
+import { GoType, GoEmitter } from './go-type';
+import { GoTypeRef } from './go-type-reference';
+import { Module } from '../module';
 
-export class TypeMapper {
-  public constructor(public value: TypeReference) {}
+export interface InterfaceField {
+  name: string;
+  parent: Interface;
+}
+
+export class InterfaceProperty implements InterfaceField {
+  public readonly name: string;
+  public readonly references?: GoTypeRef;
+  public constructor(
+    public parent: Interface,
+    public readonly property: Property,
+  ) {
+    this.name = this.property.name;
 
-  public mapType() {
-    if (this.value.type) {
-      const type = new GoType(this.value.type);
-      return type.localName;
+    if (this.property.type) {
+      this.references = new GoTypeRef(parent.parent.root, this.property.type);
     }
+  }
 
-    return this.value.toString();
+  public emit(code: CodeMaker) {
+    const name = this.property.name;
+    const type =
+      this?.references?.scopedName(this.parent.parent) ??
+      this.property.toString();
+
+    code.line(`Get${name}() ${type}`);
+    if (!this.property.protected) {
+      code.line(`Set${name}()`);
+    }
   }
 }
 
-export class Interface extends GoType {
-  public constructor(public type: InterfaceType) {
-    super(type);
+export class InterfaceMethod implements InterfaceField {
+  public readonly name: string;
+  public readonly references?: GoTypeRef;
+  public constructor(public parent: Interface, public readonly method: Method) {
+    this.name = this.method.name;
+
+    if (this.method.returns.type) {
+      this.references = new GoTypeRef(
+        parent.parent.root,
+        this.method.returns.type,
+      );
+    }
   }
 
-  public emit(code: CodeMaker): void {
-    code.openBlock(`type ${this.localName} interface`);
-    Object.values(this.type.getMethods()).forEach((method) =>
-      this.emitInterfaceMethod(code, method),
+  public emit(code: CodeMaker) {
+    const returns = this.method.returns.type.void
+      ? ''
+      : this?.references?.scopedName(this.parent.parent) ??
+        this.method.returns.toString();
+    code.line(`${code.toPascalCase(this.name)}()${returns}`);
+  }
+}
+
+export class Interface extends GoType implements GoEmitter {
```

Comment:
Can we defer changes to this? I was thinking it might make more sense to split Interfaces into two types, one of which would be datatype interfaces and only hold properties and then behavioral interfaces as a different class.